### PR TITLE
fix: Soft delete TEI, Enrollment and events

### PR DESCRIPTION
**Pull request by enricocolasante**

### CODE REVIEW

**Comment by mortenoh** [comment]
Hi, it says tests.. but this seems to also change how we delete with forceDelete in our deletion handlers, is this a requested change?

**Comment by enricocolasante** [comment]
So, this is the discussion that we should have.
Should the hard delete be possible only in the maintenance app when deleting already soft deleted items?

**Comment by mortenoh** [comment]
Well, we need a clear strategy when it comes to soft deleted objects, just asking since it changes behavior so we need to make sure we understand the implications here.

File: dhis-2/dhis-services/dhis-service-core/src/test/java/org/hisp/dhis/trackedentity/TrackedEntityInstanceServiceTest.java [hunk 1]
```diff
@@ -109,6 +169,35 @@ public void testDeleteTrackedEntityInstance()
         assertNull( entityInstanceService.getTrackedEntityInstance( teiB.getUid() ) );
     }
 
+    @Test
+    public void testDeleteTrackedEntityInstanceAndLinkedEnrollmentsAndEvents()
+    {
+        long idA = entityInstanceService.addTrackedEntityInstance( entityInstanceA1 );
+        long psIdA = programInstanceService.addProgramInstance( programInstanceA );
+        long psiIdA = programStageInstanceService.addProgramStageInstance( programStageInstanceA );
+//        programInstanceA.setId( idA );
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-core/src/test/java/org/hisp/dhis/trackedentity/TrackedEntityInstanceServiceTest.java]: Remove commented out code.

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/DefaultProgramInstanceService.java [hunk 2]
```diff
@@ -112,23 +112,7 @@ public long addProgramInstance( ProgramInstance programInstance )
...
-        {
-            programInstanceStore.delete( programInstance );
-        }
-        else
-        {
-            programInstance.setDeleted( true );
-            programInstance.setStatus( ProgramStatus.CANCELLED );
-            programInstanceStore.update( programInstance );
-        }
+        programInstanceStore.delete( programInstance );
```
Comment (mortenoh) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/DefaultProgramInstanceService.java]: Are you sure there are no side effects of removing setting the status to CANCELLED ? Maybe you can set it before you send it onwards to the deletion store?

**Pull request merged**